src/datadog/datadog_agent.cpp

```diff
@@ -184,6 +204,12 @@ nlohmann::json DatadogAgent::config_json() const {
 }
 
 void DatadogAgent::flush() {
+  OneAtATime guard{flushing_};
+  if (guard.busy()) {
```

Comment:
How can this be tested in a repeatable / reliable way without introducing additional state?
Also it has the smell of "double checked locking" edge cases, which are difficult to both prove and disprove.

Reply:
I could test it by exposing the `atomic<bool>` to a test for fiddling.

Alternatively, (and I prefer it, for some reason), I could test it "for real" by giving `DatadogAgent` an `HTTPClient` that I've mocked to block on command. Then I can play games with a mocked `EventScheduler` and the `HTTPClient` to create any interleaving of operations I want. Might need to spawn a couple threads.

By "double checked locking" edge cases, do you mean footguns in the atomic operations, or do you mean an interaction between the atomic and the mutex below it?